File: src/my_project/medium_problems/from101to150/3sum_with_multiplicity.py
from typing import List 
import collections 

class Solution:
    def threeSumMulti(self, arr: List[int], target: int) -> int:
        ans = 0
        n = len(arr)
        level2 = collections.defaultdict(int)
        for i in range(2, n):
            for j in range(i-1):
                level2[arr[j] + arr[i-1]] += 1
            ans = ans + level2[target - arr[i]]
            ans = ans % (10**9 + 7)
        return ans

File: my_project_typescript/medium_problems_typescript/from51to100/3sumWithMultiplicity.ts
function threeSumMulti(arr: number[], target: number): number {

    let ans: number = 0
    let n: number = arr.length

    let level2: Array<number> = []

    for (let i = 2; i < n; i++){
        for (let j = 0; j < i-1; j++){
            level2[arr[j]+arr[i-1]] = 0
            

        }
        level2[target-arr[i]] = 0
    }

    for (let i = 2; i < n; i++){
        for (let j = 0; j < i-1; j++){
            level2[arr[j]+arr[i-1]] += 1
        }
        ans = ans + level2[target-arr[i]] 
        ans = ans % (10**9  + 7)
    }

    return ans

};